plugin/storage/integration/elasticsearch_test.go

```diff
@@ -120,48 +117,21 @@ func (s *ESStorageIntegration) esCleanUp(t *testing.T, allTagsAsFields bool) {
 	s.initSpanstore(t, allTagsAsFields)
 }
 
-func (s *ESStorageIntegration) initSamplingStore(t *testing.T) {
-	client := s.getEsClient(t)
-	mappingBuilder := mappings.MappingBuilder{
-		TemplateBuilder: estemplate.TextTemplateBuilder{},
-		Shards:          5,
-		Replicas:        1,
-		EsVersion:       client.GetVersion(),
-		IndexPrefix:     indexPrefix,
-		UseILM:          false,
-	}
-	clientFn := func() estemplate.Client { return client }
-	samplingstore := samplingstore.NewSamplingStore(
-		samplingstore.SamplingStoreParams{
-			Client:          clientFn,
-			Logger:          s.logger,
-			IndexPrefix:     indexPrefix,
-			IndexDateLayout: indexDateLayout,
-			MaxDocCount:     defaultMaxDocCount,
-		})
-	sampleMapping, err := mappingBuilder.GetSamplingMappings()
-	require.NoError(t, err)
-	err = samplingstore.CreateTemplates(sampleMapping)
-	require.NoError(t, err)
-	s.SamplingStore = samplingstore
-}
-
-func (s *ESStorageIntegration) getEsClient(t *testing.T) eswrapper.ClientWrapper {
-	bp, err := s.client.BulkProcessor().BulkActions(1).FlushInterval(time.Nanosecond).Do(context.Background())
-	require.NoError(t, err)
-	s.bulkProcessor = bp
-	esVersion, err := s.getVersion()
-	require.NoError(t, err)
-	return eswrapper.WrapESClient(s.client, bp, esVersion, s.v8Client)
-}
-
 func (s *ESStorageIntegration) initializeESFactory(t *testing.T, allTagsAsFields bool) *es.Factory {
 	s.logger = zaptest.NewLogger(t)
 	f := es.NewFactory()
 	v, command := config.Viperize(f.AddFlags)
+	esVersion, err := s.getVersion()
```

Comment:
why is this needed? I see you pass it as an argument, but isn't it circular? When the storage is initialized it would execute the same logic to get the version: https://github.com/jaegertracing/jaeger/blob/a9a7bc22df9fd63d7eb591af2ad2c4be730b4831/pkg/es/config/config.go#L170

In fact, I would remove getVersion function from this test altogether (can be a separate PR) - the es.Client already identifies the version, we can simply expose it in its interface: https://github.com/jaegertracing/jaeger/blob/a9a7bc22df9fd63d7eb591af2ad2c4be730b4831/pkg/es/wrapper/wrapper.go#L50

Reply:
that logic was redundant. I fixed it in commit b558e9290dd16fd79d50a1a324facab632ccf1f4.
Regarding the removal of the getVersion function, are you planning to handle it as indicated in the TODO comment in the following link?
https://github.com/jaegertracing/jaeger/blob/a9a7bc22df9fd63d7eb591af2ad2c4be730b4831/plugin/storage/integration/elasticsearch_test.go#L245